### ISSUE TITLE
frida -f failed to spawn processes

### ISSUE THREAD

**Issue by progopis** [frida/frida-python#59]
I tried different apps and each time I get two types of erros, they are:
```
frida -f /usr/bin/php
    _____
   (_____)
    |   |    Frida 6.0.1 - A world-class dynamic instrumentation framework
    |   |
    |`-'|    Commands:
    |   |        help      -> Displays the help system
    |   |        object?   -> Display information about 'object'
    |   |        exit/quit -> Exit
    |   |
    |   |    More info at http://www.frida.re/docs/home/
    `._.'

Failed to spawn: unexpected error while spawning child process '/usr/bin/php' (task_for_pid returned '(os/kern) failure')
```
and

```
frida -f ./Calculator.app/Contents/MacOS/Calculator
    _____
   (_____)
    |   |    Frida 6.0.1 - A world-class dynamic instrumentation framework
    |   |
    |`-'|    Commands:
    |   |        help      -> Displays the help system
    |   |        object?   -> Display information about 'object'
    |   |        exit/quit -> Exit
    |   |
    |   |    More info at http://www.frida.re/docs/home/
    `._.'

Failed to spawn: unexpected error while resolving functions
```

Currently I don't know a reason of it.

**Comment by oleavr** [comment]
Ooh yes, this is a known issue, sorry about that. We basically spawn the process and then go looking for the libc so we can generate some early bootstrap code. This used to work but after a refactoring (or possibly an OS X update) our resolver logic no longer works. I'll investigate and push a new release as soon as I have a fix.

**Comment by oleavr** [comment]
Just released Frida 6.0.3 with a [fix](https://github.com/frida/frida-gum/commit/c1b9f2b67e31594034f84d5da1ee4c3f8e7b36a0) for this issue; hope it's working better for you now!

**Comment by progopis** [comment]
Thanks for a quick fix. I prepared new tests.
```
➜  node_modules git:(master) frida -f ./Calculator.app/Contents/MacOS/Calculator
    _____
   (_____)
    |   |    Frida 6.0.3 - A world-class dynamic instrumentation framework
    |   |
    |`-'|    Commands:
    |   |        help      -> Displays the help system
    |   |        object?   -> Display information about 'object'
    |   |        exit/quit -> Exit
    |   |
    |   |    More info at http://www.frida.re/docs/home/
    `._.'

Spawned `./Calculator.app/Contents/MacOS/Calculator`. Use %resume to let the main thread start executing!
[Local::file::['./Calculator.app/Contents/MacOS/Calculator']]-> q
Thank you for using Frida!
```
But
```
function deviceSpawn(device, path, scriptSource, clientCode, messageListener, keyHandler) {
    console.log('before spawn', path);
    device.spawn([path])
    .then(function (pid) {
        console.log('spawned:', pid);
        const target = getNameFromPid(device, pid);
        deviceAttach(device, target, scriptSource, clientCode, messageListener, keyHandler);
        return device.resume(pid);
    })
    .then(function () {
        console.log('FRAPL: process resumed');
    })
    .catch(function (error) {
        console.log('ERROR:', error.message);
    });
}
```
And I get error with this code:
```
node client.js -s ./node_modules/Calculator.app/Contents/MacOS/Calculator server.js
...
before spawn ./node_modules/Calculator.app/Contents/MacOS/Calculator
ERROR: Unexpected error while resolving functions
```
